src/main/java/edu/hm/hafner/analysis/ast/Ast.java

```diff
@@ -0,0 +1,525 @@
+     * Sets the name to the specified value.
+     *
+     * @param name
+     *         the value to set
+     */
+    public void setName(final String name) {
+        this.name = name;
+    }
+
+    /**
+     * Returns the name.
+     *
+     * @return the name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the hash code of the selected part of the AST. The hash code is derived from the SHA1 digest.
+     *
+     * @return hash code of the AST
+     */
+    public long getContextHashCode() {
+        return SHA_1_TO_LONG_CONVERTER.toLong(getDigest());
+    }
+
+    /**
+     * Returns a digest of the selected part of the AST.
+     *
+     * @return the digest of the AST, represented as SHA1 hash code
+     */
+    public String getDigest() {
+        List<DetailAST> elements = chooseArea();
+        StringBuilder astElements = new StringBuilder();
+        children.clear();
+        for (DetailAST astElement : elements) {
+            int type = astElement.getType();
+
+            if (type == TokenTypes.TYPE) {
+                children = calcAllChildren(astElement.getFirstChild());
+                for (DetailAST child : children) {
+                    astElements.append(child.getText());
+                    astElements.append(DELIMITER);
+                }
+                children.clear();
+            }
+
+            boolean lockedNextElement = false;
+            if (!constants.isEmpty()) {
+                for (DetailAST ast : constants.keySet()) {
+                    if (ast.getType() == TokenTypes.IDENT && ast.getText().equals(astElement.getText())) {
+                        astElements.append(TokenUtil.getTokenName(constants.get(ast).getType()));
+                        astElements.append(DELIMITER);
+                        lockedNextElement = true;
+                    }
+                }
+            }
+            if (type != TokenTypes.TYPE && !lockedNextElement) {
+                astElements.append(TokenUtil.getTokenName(type));
+                astElements.append(DELIMITER);
+            }
+        }
+
+        if (getName() != null) {
+            astElements.append(name);
+        }
+
+        return createHashCodeFromAst(astElements);
+    }
+
+    private String createHashCodeFromAst(final StringBuilder astElements) {
+        MessageDigest messageDigest = createMessageDigest();
+        byte[] digest = messageDigest.digest(astElements.toString().getBytes(Charset.forName(CHARSET)));
+
+        StringBuilder result = new StringBuilder();
+        for (byte b : digest) {
+            result.append(String.format("%02x", b));
+        }
+        return result.toString();
+    }
+
+    private MessageDigest createMessageDigest() {
+        try {
+            return MessageDigest.getInstance(HASH_ALGORITHM);
+        }
+        catch (NoSuchAlgorithmException e) {
+            throw new IllegalArgumentException("Requested algorithm not found: " + HASH_ALGORITHM);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [New exception is thrown in catch block, original stack trace may be lost](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)